components/director/hack/plugins/descriptionsdecorator/plugin.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/vektah/gqlparser/ast"
+)
+
+type GraphqlOperationType string
+
+const (
+	Query             GraphqlOperationType = "query"
+	Mutation          GraphqlOperationType = "mutation"
+	ExamplesDirectory                      = "../../examples"
+	UnsanitizedAPI                         = "A-P-I"
+)
+
+var _ plugin.ConfigMutator = &descriptionsDecoratorPlugin{}
+
+func NewPlugin(schemaFileName string) *descriptionsDecoratorPlugin {
+	return &descriptionsDecoratorPlugin{schemaFileName: schemaFileName}
+}
+
+type descriptionsDecoratorPlugin struct {
+	schemaFileName string
+}
+
+func (p *descriptionsDecoratorPlugin) Name() string {
+	return "descriptions_decorator"
+}
+
+func (p *descriptionsDecoratorPlugin) MutateConfig(cfg *config.Config) error {
+	fmt.Printf("[%s] Mutate Configuration\n", p.Name())
+
+	if err := cfg.Check(); err != nil {
+		return err
+	}
+
+	schema, _, err := cfg.LoadSchema()
+	if err != nil {
+		return err
+	}
+
+	for _, f := range schema.Query.Fields {
+		p.ensureDescription(f, Query)
+	}
+
+	for _, f := range schema.Mutation.Fields {
+		p.ensureDescription(f, Mutation)
+	}
+
+	if err := cfg.Check(); err != nil {
+		return err
+	}
+
+	schemaFile, err := os.Create(p.schemaFileName)
+	if err != nil {
+		return err
+	}
+
+	f := plugins.NewFormatter(schemaFile)
+	f.FormatSchema(schema)
+	return schemaFile.Close()
+}
+
+func (p *descriptionsDecoratorPlugin) ensureDescription(f *ast.FieldDefinition, opType GraphqlOperationType) {
+
+	f.Description = deletePrevious(f.Description)
+	dirs, err := ioutil.ReadDir(ExamplesDirectory)
+	if err != nil {
+		log.Printf("While reading the examples directory: %s", err.Error())
+		return
+	}
+	for _, dir := range dirs {
+		if !dir.IsDir() {
+			continue
+		}
+		if sanitizeName(f.Name, opType) != dir.Name() {
+			continue
+		}
+		files, err := ioutil.ReadDir(path.Join(ExamplesDirectory, dir.Name()))
+		if err != nil {
+			log.Printf("While reading the examples subdirectory %s : %s", dir.Name(), err.Error())
+			return
+		}
+		for _, file := range files {
+			f.Description = addExample(f.Description, dir.Name(), file.Name())
+		}
+
+	}
+}
+
+func sanitizeName(name string, opType GraphqlOperationType) string {
+	counter := 0
+	for index, letter := range name {
+		if unicode.IsUpper(letter) {
+			if index == 0 {
+				continue
+			}
+			name = fmt.Sprintf("%s-%s", name[:index+counter], name[index+counter:])
+			counter++
+		}
+	}
+	if strings.Contains(name, UnsanitizedAPI) {
+		name = strings.ReplaceAll(name, UnsanitizedAPI, "api")
+	}
+
+	if opType == Query {
+		return strings.ToLower(fmt.Sprintf("query-%s", name))
+	}
+	return strings.ToLower(name)
+
+}
+
+func deletePrevious(description string) string {
+	if len(description) == 0 {
+		return ""
+	}
+	if description[:3] == "see" {
+		return ""
+	}
+	index := strings.Index(description, "\\")
+	if index == -1 {
+		return description
+	}
+	return description[:index]
+}
+
+func addExample(description string, dirName string, name string) string {
```

Comment:
It doesn't look that fancy:

![Screen Shot 2019-11-06 at 12 18 57](https://user-images.githubusercontent.com/7155799/68294116-a8c49400-008f-11ea-8b14-b03feb115418.png)

Instead of:

```
"""
	see example [here](examples/create-application/create-application-with-apis.graphql)\
	 see example [here](examples/create-application/create-application-with-documents.graphql)\
	 see example [here](examples/create-application/create-application-with-event-apis.graphql)\
	 see example [here](examples/create-application/create-application-with-webhooks.graphql)\
	 see example [here](examples/create-application/create-application.graphql)
"""
```

Please do it this way:
```
	"""
	Lorem ipsum

	**Examples**
	- [create application with apis](examples/create-application/create-application-with-apis.graphql)
	- [create application with documents](examples/create-application/create-application-with-documents.graphql)
	- [create application with event apis](examples/create-application/create-application-with-event-apis.graphql)
	- [create application with webhooks](examples/create-application/create-application-with-webhooks.graphql)
	- [create application](examples/create-application/create-application.graphql)
	"""
```

Result:

![Screen Shot 2019-11-06 at 12 34 29](https://user-images.githubusercontent.com/7155799/68295137-02c65900-0092-11ea-9ea1-1193d62accb9.png)